### PR TITLE
Support more Android arch in Makefile build

### LINKED ISSUE #11996: tensorflow Makefile build does not support building arm64-v8a static library for Android
Currently tensorflow has only support build static library in `armeabi-v7a` for Android.  

And the build script has not support the other platforms yet as we can see in:
 [`tensorflow/tensorflow/contrib/makefile/Makefile`](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/makefile/Makefile#L219)

Our team would like to build all the architecture for Android if possible, and we tried to fixed the `makefile` that expect to link with our Android project but got many linking errors except `armeabi-v7a`.

So, I was just wondering if tensorflow will support those platform: `arm64-v8a`, `x86`, `x88_64`?

### PR DESCRIPTION
**Pull request by resec**
Adds support for various Android arch in Makefile build.

This resolves #11996

### CODE REVIEW

**Comment by resec** [comment]
This pr is ready for review

**Comment by drpngx** [comment]
Jenkins, test this please.

@andrewharp any opinion on this?

**Comment by sb2nov** [comment]
@andrewharp ping?

File: tensorflow/contrib/makefile/Makefile [hunk 1]
```diff
@@ -222,43 +226,93 @@ ifeq ($(TARGET),ANDROID)
 # NDK_ROOT=/path/to/your/ndk
 # You need to have an Android version of the protobuf libraries compiled to link
 # in. The compile_android_protobuf.sh script may help.
-# TODO(satok): Support all CPU architectures (Currently only armv7 is supported)
 
-	OS_PATH :=
+	ANDROID_OS_ARCH :=
```
Comment (andrewharp) [inline tensorflow/contrib/makefile/Makefile]: ANDROID_HOST_OS_ARCH would be more explicit

File: tensorflow/contrib/makefile/build_all_android.sh [hunk 2]
```diff
@@ -32,13 +35,16 @@ if [[ -z "${NDK_ROOT}" ]]; then
     exit 1
 fi
 
-while getopts "Es:t:Tx:" opt_name; do
+ARCH=armeabi-v7a
```
Comment (andrewharp) [inline tensorflow/contrib/makefile/build_all_android.sh]: What is the exact difference between this and ANDROID_ARCH?

Comment (resec) [reply tensorflow/contrib/makefile/build_all_android.sh]: They are the same, this is a short name as we are inside the `build_all_android.sh`.

**Comment by resec** [comment]
@andrewharp, already changed ANDROID_OS_ARCH to ANDROID_HOST_OS_ARCH, thanks.

**Review by andrewharp — APPROVED** [review]

**Comment by andrewharp** [comment]
@lihanchen FYI

**Comment by andrewharp** [comment]
Jenkins, test this please.

**Comment by andrewharp** [comment]
@benoitsteiner can this be merged? I don't know why the checks don't seem to finish.

**Comment by andrewharp** [comment]
Jenkins, test this please.

**Comment by andrewharp** [comment]
@gunan Any idea what Ubuntu CC never reports back in?

**Comment by gunan** [comment]
Those are the new tests, we need to trigger them using "kokoro:force-run" tag

**Comment by andrewharp** [comment]
Jenkins, test this please.

**Comment by drpngx** [comment]
```
ERROR: Skipping '//tensorflow/contrib/eager/python:datasets_test': no such target '//tensorflow/contrib/eager/python:datasets_test': target 'datasets_test' not declared in package 'tensorflow/contrib/eager/python' defined by /tmpfs/src/github/tensorflow/tensorflow/contrib/eager/python/BUILD
```

Not sure. Re-trying.

**Pull request merged**